spec/currency_spec.rb

```diff
@@ -37,6 +37,65 @@
     it "raises when the currency is nil" do
       expect { Money::Currency.new(nil) }.to raise_error(Money::Currency::UnknownCurrency)
     end
+    
+    it "looks up crypto currencies when enabled" do
+      config = Money::Config.new.tap do |config|
+        config.experimental_crypto_currencies = true
+      end
+      
+      allow(Money::Currency).to receive(:currencies).and_return({})
+      crypto_currencies = { 
+        "usdc" => { 
+          "iso_code" => "USDC", 
+          "name" => "USD Coin",
+          "symbol" => "USDC",
+          "disambiguate_symbol" => "USDC",
+          "subunit_to_unit" => 100,
+          "smallest_denomination" => 1,
+          "decimal_mark" => "."
+        } 
+      }
+      allow(Money::Currency).to receive(:crypto_currencies).and_return(crypto_currencies)
+      
+      with_config(config) do
+        currency = Money::Currency.new('USDC')
+        expect(currency.iso_code).to eq('USDC')
+        expect(currency.symbol).to eq('USDC')
+      end
+    end
+    
+    it "doesn't look up crypto currencies when disabled" do
+      # Reset class variables to ensure clean state
+      old_currencies = Money::Currency.class_variable_get(:@@loaded_currencies) rescue {}
+      Money::Currency.class_variable_set(:@@loaded_currencies, {})
+      
+      # Explicitly set config
+      old_config = Money.config
+      Money.config = Money::Config.new
+      expect(Money.config.experimental_crypto_currencies).to eq(false)
+      
+      # Mock currencies
+      allow(Money::Currency).to receive(:currencies).and_return({})
+      crypto_currencies = { 
+        "usdc" => { 
+          "iso_code" => "USDC", 
+          "name" => "USD Coin",
+          "symbol" => "USDC",
+          "disambiguate_symbol" => "USDC",
+          "subunit_to_unit" => 100,
+          "smallest_denomination" => 1,
+          "decimal_mark" => "."
+        } 
+      }
+      allow(Money::Currency).to receive(:crypto_currencies).and_return(crypto_currencies)
+      
+      # When crypto_currencies is disabled, new should raise an error for a crypto currency
+      expect { Money::Currency.new('USDC') }.to raise_error(Money::Currency::UnknownCurrency)
+      
+      # Cleanup
+      Money.config = old_config
```

Comment:
question: Can/should we use `with_config` here? Same in the test below.

```suggestion
      config = Money::Config.new
      config.experimental_crypto_currencies = false

      with_config(config) do
        # Mock currencies
        allow(Money::Currency).to receive(:currencies).and_return({})
        crypto_currencies = { 
          "usdc" => { 
            "iso_code" => "USDC", 
            "name" => "USD Coin",
            "symbol" => "USDC",
            "disambiguate_symbol" => "USDC",
            "subunit_to_unit" => 100,
            "smallest_denomination" => 1,
            "decimal_mark" => "."
          } 
        }
        allow(Money::Currency).to receive(:crypto_currencies).and_return(crypto_currencies)

        # When crypto_currencies is disabled, new should raise an error for a crypto currency
        expect { Money::Currency.new('USDC') }.to raise_error(Money::Currency::UnknownCurrency)
      end

```